src/Form/PrepaidBalanceReportsDownloadForm.php

```diff
@@ -0,0 +1,230 @@
+      ];
+
+      return $form;
+    }
+
+    if (!count($this->billingDocuments)) {
+      $form['year'] = [
+        '#markup' => $this->t('There are no billing documents for your account.'),
+      ];
+
+      return $form;
+    }
+
+    // Build currency options.
+    $currency_options = [];
+    foreach ($this->supportedCurrencies as $currency) {
+      $currency_options[$currency->id()] = "{$currency->getName()} ({$currency->getDisplayName()})";
+    }
+
+    // Save this to form state to be available in submit callback.
+    $form_state->set('currency_options', $currency_options);
+
+    $form['currency'] = [
+      '#title' => $this->t('Select an account'),
+      '#type' => 'select',
+      '#required' => TRUE,
+      '#options' => [
+        '' => $this->t('Select an account'),
+      ] + $currency_options,
+    ];
+
+    // Build date options.
+    $date_options = [];
+    array_map(function ($document) use (&$date_options) {
+      $date_options[$document->year][$document->year . '-' . $document->month] = ucwords(strtolower($document->monthEnum));
+    }, $this->billingDocuments);
+
+    // Save this to form state to be available in submit callback.
+    $form_state->set('date_options', $date_options);
+
+    $form['year'] = [
+      '#title' => $this->t('Select a year'),
+      '#type' => 'select',
+      '#required' => TRUE,
+      '#options' => [
+        '' => $this->t('Select a year'),
+      ] + array_combine(array_keys($date_options), array_keys($date_options)),
+    ];
+
+    foreach ($date_options as $year => $date_option) {
+      $form["month_$year"] = [
+        '#title' => $this->t('Select a month'),
+        '#type' => 'select',
+        '#options' => [
+          '' => $this->t('Select a month'),
+        ] + $date_option,
+        '#states' => [
+          'required' => [
+            'select[name="year"]' => ['value' => $year],
+          ],
+          'visible' => [
+            'select[name="year"]' => ['value' => $year],
+          ]
+        ],
+      ];
+    }
+
+    $form['actions'] = [
+      '#type' => 'actions',
+    ];
+
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Download CSV'),
+      '#button_type' => 'primary',
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    if (($year = $form_state->getValue('year')) && ($currency = $form_state->getValue('currency')) && ($date = $form_state->getValue("month_$year"))) {
+      try {
+        $billing_date = new \DateTimeImmutable($date);
+
+        if ($report = $this->monetization->getPrepaidBalanceReports($this->user->getEmail(), $billing_date, $currency)) {
+          $filename = "prepaid-balance-report-$date.csv";
+          $response = new Response($report);
+          $response->headers->set('Content-Type', 'text/csv');
+          $response->headers->set('Content-Disposition', "attachment; filename='$filename'");
```

Comment:
On Firefox 64, macOS 10.14.1, this causes the file to be saved with quotes around it, i.e. `'example.csv'`.

Reply:
Fixed.